### PR TITLE
added subscription owner address to OracleRequest

### PR DESCRIPTION
**Pull request by KuphJr**

### CODE REVIEW

File: contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol
Comment (pinebit) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:17]: This change would require a change in Atlas. Because of different ABI.

**Comment by pinebit** [comment]
> Go solidity wrappers are out-of-date, regenerate them via the `make go-solidity-wrappers` command

Don't miss this.

**Review by bolekk — CHANGES_REQUESTED** [review]
Can we hold off for a bit until we discuss what other metadata needs to go there.

File: contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol
Comment (KuphJr) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:14]: Add to this event:
`address requestingContract`,
`address requestInitiator`,

Comment (KuphJr) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:14]: emit:

tx.origin & msg.sender

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (bolekk) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol:833]: We shouldn't use "whitelist" as a name of anything.

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (bolekk) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol:636]: Doesn't need to external, right?

**Review by bolekk — CHANGES_REQUESTED** [review]
LGTM overall but I want others to look at it, especially Justin.

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (KuphJr) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol:636]: It does since it is called by OCR2DROracle which is an external contract.

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (KuphJr) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol]: changed modifer name to `onlyAuthorizedUsers`

**Review by bolekk — APPROVED** [review]

File: contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol
Comment (justinkaseman) [inline contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol:92]: What if we made this fetch the whole subscription instead? In case we ever need to get something off of it later

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (justinkaseman) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol:25]: I thought the plan was to whitelist subscriptions? Why is this an oracle?

**Comment by justinkaseman** [comment]
Also, going to need to fix tests on this PR before merging

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (justinkaseman) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol:831]:
```suggestion
  /**
   * @dev The allow list is kept on the DON contract. This modifier checks if a user is authorized from there.
   */
```

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (bolekk) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol:833]: In the oracle contract, we authorize users' wallets, while here we check is a contract is authorized. Is this wrong or am I missing something?

File: contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol
Comment (KuphJr) [reply contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol:92]: Do you think this is worth adding now or can we make this change later if needed?

I would advocate for not worrying about this now as the OracleRequest event may change more prior to mainnet per our discussion yesterday.

File: contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol
Comment (KuphJr) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol:835]: This was discussed via DM, but is documented here for clarity.

What prevents this is `tx.origin` is used to gate sendRequest while `msg.sender` is used to gate createSubscription.

Therefore, in the case described above, the Automation node's wallet would not be calling `createSubscrption`, the Automation node contract would be making the call and that call would fail since the Automation node's contract is not in the allowList.

File: core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go
Comment (bolekk) [inline core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go:343]: The "event" keyword shouldn't be here.

**Review by pinebit — APPROVED** [review]

File: contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol
Comment (justinkaseman) [reply contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol:92]: Sounds good to me. I'll add a new Jira ticket so that we remember.

**Pull request merged**
